autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -472,8 +486,29 @@ impl BigTableClientImpl {
         let mut req = DropRowRangeRequest::new();
         req.set_name(self.settings.table_name.clone());
         req.set_row_key_prefix(row_key.as_bytes().to_vec());
+        // Admin calls use a slightly different routing param and a truncated prefix
+        // See https://github.com/googleapis/google-cloud-cpp/issues/190#issuecomment-370520185
+        let prefix = &self
+            .settings
+            .table_name
+            .split("/tables/")
+            .collect::<Vec<&str>>()
+            .first()
+            .copied();
+        if prefix.is_none() {
+            return Err(error::BigTableError::Admin(
+                "Invalid table name specified".to_string(),
+                None,
+            ));
+        }
+        let admin_meta = MetadataBuilder::with_prefix(prefix.unwrap())
```

Comment:
Let's move all this to the constructor

Also you can just `next()` on the iterator here:
```suggestion
        let prefix = self
            .settings
            .table_name
            .split("/tables/")
            .next();
        let Some(prefix) = prefix else {
            return Err(error::BigTableError::Admin(
                "Invalid table name specified".to_string(),
                None,
            ));
        };
        let admin_meta = MetadataBuilder::with_prefix(prefix)
```

But I think `split_once` which was added a few years ago is even better (I like that it's a bit more strict, returning `None` when there's no match instead of producing empty strings):

```suggestion
        let Some((prefix, _)) = self.settings.table_name.split_once("/tables/") else {
            return Err(error::BigTableError::Admin(
                "Invalid table name specified".to_string(),
                None,
            ));
        };
        let admin_meta = MetadataBuilder::with_prefix(prefix)
```